papers/bcr-2020-015-account.md

```diff
@@ -0,0 +1,351 @@
+ - Script type
+ - Fingerprint for the master public key (see [BIP32])
+ - Derivation path from the master key to the account level
+ - The xpub at that derivation
+
+This information can be contained in an output descriptor [OD-IN-CORE], which is defined in CBOR with [BCR-2020-010] and represented in UR using the ``ur:crypto-output`` type. 
+Using this format, each output descriptor defines a single script type, derivation path and parent fingerprint (``crypto-keypath``) and deterministic key (``crypto-hdkey``).
+
+There is a common need however to share information for multiple script types. 
+For example, a user may define the script type on wallet software for which the relevant xpub must be retrieved from a hardware wallet.
+This suggests that a number of output descriptors for standardized script types could be packaged into a CBOR data format.
+Doing so removes the need for the user to also correctly define the matching script type when exporting from the hardware wallet.
+The user would only (optionally) need to specify the account number.
+In addition, this would promote standardization in derivation paths between different wallets.
+
+This document specifies a native CBOR encoding for a number of standardized script types along with their common derivations to the account level. 
+This encoding can be shared as a Uniform Resource [UR] type `crypto-account` (CBOR tag #6.311) for transmitting account level data.
+
+In the encoding, each script type is represented in an output descriptor. 
+These output descriptors are then packaged into an array. 
+Note that in order to include the parent fingerprint for each ``crypto-hdkey``, the master fingerprint (which is common to all accounts) is included separately.
+The top level packaging is a map of the master fingerprint, and the array of output descriptors.
+
+The following standardized script types may be present in a ``crypto-account`` encoding, shown here with default derivations for Bitcoin mainnet and account #0:
+
+| Script type | Default Derivation |
+| ----------- | ---------- |
+| P2PKH | ``m/44'/0'/0'`` |
+| P2SH-P2WPKH | ``m/49'/0'/0'`` |
+| P2WPKH | ``m/84'/0'/0'`` |
+| Multisig P2SH | ``m/45'`` |
+| Multisig P2SH-P2WSH | ``m/48'/0'/0'/1'`` |
+| Multisig P2WSH | ``m/48'/0'/0'/2'`` |
+
+If the software creating the encoding does not support a particular script type it should be omitted. 
+Note that legacy multisig with P2SH does not support different networks or accounts - as per [BIP45] the hardened derivation path is only one level of `m/45'`. 
+
+### CDDL
+
+The following specification is written in Concise Data Definition Language [CDDL].
+
+```
+; Output descriptors here are restricted to HD keys at account level key derivations only (no 0/* or 1/* children crypto-keypaths)
+
+output_exp = #6.308(crypto-output)
+
+account = {
+    1: uint32, ; Master fingerprint (fingerprint for the master public key as per BIP32)
```

Comment:
Please define map keys symbolically and define their integer values below. See, for example [here](https://github.com/BlockchainCommons/Research/blob/master/papers/bcr-2020-010-output-desc.md#cddl).

Reply:
Done, thanks for the pointer.